Source/Core/VideoCommon/TextureCacheBase.cpp

```diff
@@ -297,6 +295,10 @@ TextureCacheBase::TCacheEntryBase*
 TextureCacheBase::DoPartialTextureUpdates(TCacheEntryBase* entry_to_update, u8* palette,
                                           u32 tlutfmt)
 {
+  if (!entry_to_update->may_have_overlapping_textures)
+    return entry_to_update;
+  entry_to_update->may_have_overlapping_textures = false;
```

Comment:
This condition should be documented, maybe something like: "If there is no chance of overlapping textures or we already applied the partial texture updates, we can immediately return."